src/FileUploader/FileUploader.svelte

```diff
@@ -95,11 +95,11 @@
     kind="{kind}"
     on:change
     on:change="{({ target }) => {
-      files = [...target.files].map(({ name }) => name);
+      files = [...target.files];
     }}"
   />
   <div class:bx--file-container="{true}">
-    {#each files as name, i (name)}
+    {#each files as { name }, i (name)}
```

Comment:
... but if the `FileList` type is kept intact, the rendering logic would need to be updated, which expects `files` to be an array

Reply:
This is not only the logic to update, but also at anywhere which use files as an array, such as   [L71](https://github.com/AsakuraMizu/carbon-components-svelte/blob/410a805191350e9537d1f77455cf76536291fa87/src/FileUploader/FileUploader.svelte#L71)(`files.includes`), [L112](https://github.com/AsakuraMizu/carbon-components-svelte/blob/410a805191350e9537d1f77455cf76536291fa87/src/FileUploader/FileUploader.svelte#L112) and [L117](https://github.com/AsakuraMizu/carbon-components-svelte/blob/410a805191350e9537d1f77455cf76536291fa87/src/FileUploader/FileUploader.svelte#L117)(`files.filter`). So maybe changing its type to `File[]` is a better solution.

Reply:
@AsakuraMizu makes sense – another point is that the end user probably expects an array type anyway.